terraform/aptos-node/gcp/dns.tf

```diff
@@ -56,9 +56,9 @@ resource "google_dns_record_set" "fullnode" {
 }
 
 output "validator_endpoint" {
-  value = var.zone_name != "" ? "/dns4/${trimsuffix(google_dns_record_set.validator[0].name, ".")}/tcp/${data.kubernetes_service.validator-lb[0].spec[0].port[0].port}" : null
+  value = var.zone_name != "" && var.create_records ? "/dns4/${trimsuffix(google_dns_record_set.validator[0].name, ".")}/tcp/${data.kubernetes_service.validator-lb[0].spec[0].port[0].port}" : null
```

Comment:
This access path seems error prone if the service isnt present?

Reply:
yea we're kinda making the assumption that the k8s services come up without issue. no way around this imo since we're still coupling infra and application release. At least it will fail with a good error